### PR TITLE
Homeworks to the Lectures on Pytest and on HTTP

### PR DESCRIPTION
**Pull request by denys-lozinskyi**
Привет!

Проревьювь, пожалуйста, мои домашки.
По HTTP - в Lecture_3/Homework
По Pytest - в папке Lecture_Pytest.
Буду очень признателен за фидбэк.

### CODE REVIEW

File: Denys_Lozinskyi/Lecture_3/Homework/homework_1.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import requests
+from requests.exceptions import HTTPError
+
+URL = 'http://0.0.0.0:5002/'
+endpoints = ['', 'fail500', 'basic_auth']
```
Comment (SergiiButenko) [inline Denys_Lozinskyi/Lecture_3/Homework/homework_1.py]: '' - what is that?

File: Denys_Lozinskyi/Lecture_3/Homework/homework_1.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+for endpoint in endpoints:
+    url = URL + endpoint
```
Comment (SergiiButenko) [inline Denys_Lozinskyi/Lecture_3/Homework/homework_1.py]: please take a look on URL join methods

File: Denys_Lozinskyi/Lecture_3/Homework/homework_1.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    try:
+        print('\nSending GET request to {}'.format(url))
```
Comment (SergiiButenko) [inline Denys_Lozinskyi/Lecture_3/Homework/homework_1.py]: use formatted string

File: Denys_Lozinskyi/Lecture_3/Homework/homework_2.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+
+def create_resource(item: dict) -> str:
+    """ Creates a resource with a 'name' provided
+        :param item: name of the resource to be created
+        :return :int: the ID of the item created
+    """
+    try:
+        response = requests.post(
+            URL + 'items', json=item, headers={'Authorization': f'Bearer {access_token}'}
```
Comment (SergiiButenko) [inline Denys_Lozinskyi/Lecture_3/Homework/homework_2.py]: use var

File: Denys_Lozinskyi/Lecture_Pytest/homework/tests/test_suite.py [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    os.killpg(process.pid, signal.SIGTERM)
+
+
+@pytest.fixture(params=testdata.correct_credentials)
+def authorization_positive(request):
+    """ use as a SETUP for tests require preliminary authorization with access token """
+    user_name, password = request.param['user_name'], request.param['password']
+    response_1 = requests.post(testdata.url + 'login', json={'username': user_name, 'password': password})
+    testdata.access_token = response_1.json()['access_token']
+    assert len(testdata.access_token) > 0, "Token not received"
```
Comment (SergiiButenko) [inline Denys_Lozinskyi/Lecture_Pytest/homework/tests/test_suite.py]: redo

File: Denys_Lozinskyi/Lecture_Pytest/homework/tests/test_suite.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    # logging in with the access token received
+    response_2 = requests.get(testdata.url + 'protected', headers={'Authorization': f'Bearer {testdata.access_token}'})
+    assert 200 <= response_2.status_code < 300
```
Comment (SergiiButenko) [inline Denys_Lozinskyi/Lecture_Pytest/homework/tests/test_suite.py]: raise_for_status

**Review by SergiiButenko — CHANGES_REQUESTED** [review]
Fix issues

File: Denys_Lozinskyi/Lecture_3/Homework/homework_1.py (hunk 1 above)
Comment (denys-lozinskyi) [reply Denys_Lozinskyi/Lecture_3/Homework/homework_1.py]: Resolved in 6aaa4df commit

File: Denys_Lozinskyi/Lecture_3/Homework/homework_1.py (hunk 2 above)
Comment (denys-lozinskyi) [reply Denys_Lozinskyi/Lecture_3/Homework/homework_1.py]: Used urljoin lib. Resolved in 6aaa4df commit.

File: Denys_Lozinskyi/Lecture_3/Homework/homework_1.py (hunk 3 above)
Comment (denys-lozinskyi) [reply Denys_Lozinskyi/Lecture_3/Homework/homework_1.py]: Taking full advantage of f-strings in 6aaa4df commit

File: Denys_Lozinskyi/Lecture_3/Homework/homework_2.py (hunk 4 above)
Comment (denys-lozinskyi) [reply Denys_Lozinskyi/Lecture_3/Homework/homework_2.py]: Done

File: Denys_Lozinskyi/Lecture_Pytest/homework/tests/test_suite.py (hunk 5 above)
Comment (denys-lozinskyi) [reply Denys_Lozinskyi/Lecture_Pytest/homework/tests/test_suite.py]: Done

File: Denys_Lozinskyi/Lecture_Pytest/homework/tests/test_suite.py (hunk 6 above)
Comment (denys-lozinskyi) [reply Denys_Lozinskyi/Lecture_Pytest/homework/tests/test_suite.py]: Can't see how raise_for_status can be introduced into this particular case. Encapsulated the method instead.

**Comment by denys-lozinskyi** [comment]
Introduced all changes in 6aaa4df commit

File: Denys_Lozinskyi/Lecture_Pytest/homework/config.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+local_link_to_server = 'homework/server/simple-app.py'
+
+# URLs
+base_url = 'http://0.0.0.0:5002'
```
Comment (SergiiButenko) [inline Denys_Lozinskyi/Lecture_Pytest/homework/config.py:4]: should be upper case

**Review by SergiiButenko — APPROVED** [review]

File: Denys_Lozinskyi/Lecture_Pytest/homework/config.py (hunk 7 above)
Comment (denys-lozinskyi) [reply Denys_Lozinskyi/Lecture_Pytest/homework/config.py:4]: Done

**Pull request merged**